File: Node.js/sample_listener.js
var fs=require("fs");

var Communicator = require("./Library/Communicator/index.js");

var communicator = new Communicator();

communicator.config({password: "youcanchangethis" });

communicator.listen(1234).then(() =>
{
    communicator.on("TestMessage", (obj) =>
    {
        console.log("A program from "+obj.lang+" "+obj.langversion+" on "+obj.os+" "+obj.osversion+" connected");
        if(obj.senderimg)
        {
        fs.writeFileSync(__dirname+"/ReceivedImage.png",obj.senderimg);
        console.log("An image was received and saved to: ReceivedImage.png");
        }
        if(obj.msg)
        {
        console.log("Received a message:");
        console.log(obj.msg);
        }
    });
    
});

File: Python/sample_listener.py
import imp
import multiprocessing
import time
import communicator
import csv
import os.path

def onmsg(obj):

  print "A program from "+obj["lang"]+" "+obj["langversion"]+" on "+obj["os"]+" "+obj["osversion"]+" connected"

  if(obj["senderimg"]):
    newimg=open("ReceivedImage.png","wb")
    newimg.write(obj["senderimg"])  
    print ("An image was received and saved to: ReceivedImage.png")
  
  if(obj["msg"]):
    print ("Received a message:")
    print (obj["msg"])
  
communicator = communicator.Communicator("youcanchangethis")


communicator.on("TestMessage",onmsg)
communicator.listen(1234)

input("Listening.. \n")